### PR TITLE
feat(solver/app): add cursor db

### PR DESCRIPTION
**Pull request by corverroos**
Adds a cursor DB to solver .

issue: #2384

### CODE REVIEW

File: solver/app/cursor.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	}
+
+	return &cursors{
+		table: dbStore.CursorTable(),
+	}, nil
+}
+
+// cursors provides a thread-safe persisted cursor store.
+type cursors struct {
+	mu    sync.Mutex
```
Comment (chmllr) [inline solver/app/cursor.go:57]: Do we have to exclusively lock on reads?

**Review by chmllr — APPROVED** [review]

File: solver/app/cursor.go (hunk 1 above)
Comment (kc1116) [inline solver/app/cursor.go:57]:
```suggestion
	mu    sync.RWMutex
```

File: solver/app/cursor.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	table CursorTable
+}
+
+// Get returns the block height of the cursor for the given chain version, or false, or an error.
+func (c *cursors) Get(ctx context.Context, chainVer xchain.ChainVersion) (uint64, bool, error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
```
Comment (kc1116) [inline solver/app/cursor.go:64]:
```suggestion
	c.mu.RLock()
	defer c.mu.RUnlock()
```

File: solver/app/cursor.go (hunk 1 above)
Comment (corverroos) [reply solver/app/cursor.go:57]: to be safe yes, the cosmosDB is a single threaded thing, I'm not sure concurrent reads are supported or not. We only read from the DB on startup in any case.

Comment (corverroos) [reply solver/app/cursor.go:57]: You are assuming this is safe? how do you know this?

**Pull request merged**